distributed/tests/test_steal.py

```diff
@@ -1142,8 +1148,8 @@ def block(x, event):
 
     event = Event()
     futs1 = [
-        c.submit(block, f, event=event, key=f"f1-{ix}")
-        for f in roots
+        c.submit(block, r, event=event, key=f"f{ir}-{ix}")
+        for ir, r in enumerate(roots)
```

Comment:
@gjoseph92 this is why the test was flaky with queueing: there were key collisions. The problem was actually unrelated to queueing and just a matter of subtle timings; it became flaky with AMM too.